crates/matrix-sdk-ui/src/timeline/inner.rs

```diff
@@ -93,6 +96,45 @@ pub(super) struct TimelineInnerState {
         HashMap<OwnedUserId, HashMap<ReceiptType, (OwnedEventId, Receipt)>>,
 }
 
+pub(super) struct TimelineInnerSettings {
+    pub(super) track_read_receipts: bool,
+    pub(super) event_filter: Box<TimelineEventFilterFn>,
+}
+
+impl fmt::Debug for TimelineInnerSettings {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("TimelineInnerSettings")
+            .field("track_read_receipts", &self.track_read_receipts)
+            .finish_non_exhaustive()
+    }
+}
+
+impl Default for TimelineInnerSettings {
+    fn default() -> Self {
+        Self { track_read_receipts: false, event_filter: Box::new(default_timeline_event_filter) }
+    }
+}
+
+pub(super) type TimelineEventFilterFn =
+    dyn Fn(&Raw<AnySyncTimelineEvent>, Option<&AnySyncTimelineEvent>) -> bool + Send + Sync;
+
+pub(super) fn default_timeline_event_filter(
```

Comment:
This reflects the current behavior, afaict. But we could change it to e.g. not show FailedToParse or OtherState, depending on Element X's behavior.

Reply:
I think the default filter should just be `|_, _| true`. Also, I'm wondering if the raw event is really all that useful for filtering. Do you need it for Fractal? I'm wondering if it would potentially make more sense to only allow filtering of successfully-deserialized events, and add different settings for failed-to-parse items and such, if needed at all.

Reply:
What about users sending custom events?

Reply:
They might want to show their custom events but hide others

Reply:
Even if we do want to filter them out for EX (seems likely), I'd rather add an optional filter for those that EX can use, instead of filtering out anything by default.

Reply:
Custom events won't fail deserialization :)

Reply:
I.e. they can decide based on the [`event_type`](https://docs.ruma.io/ruma/events/enum.AnySyncTimelineEvent.html#method.event_type).

Reply:
Ah right. Okay I'm gonna change that.